### ISSUE TITLE
Unable to deploy contract for unknown reasons

### ISSUE THREAD

**Issue by unparalleled-js** [ApeWorX/ape#293]
### Environment information

Latest ape
python 3.9.7
macos

### What went wrong?

tried to deploy `ERC721` contract by itself (from open zeppelin)
ran into this error:

```
~/PycharmProjects/ape/src/ape_ethereum/ecosystem.py in encode_calldata(self, abi, *args)
    232         if abi.inputs:
    233             input_types = [i.canonical_type for i in abi.inputs]
--> 234             return abi_encode(input_types, args)
    235 
    236         else:

....

ValueOutOfBounds: Value `()` of type <class 'tuple'> cannot be encoded by TupleEncoder: value has 0 items when 2 were expected
```

### How can it be fixed?

Verify args and inputs first? not sure why it is failing

**Comment by fubuloubu** [comment]
seems like it should be `if len(abi.inputs) > 0`

**Comment by unparalleled-js** [comment]
Incidentally fixed by #313 !
